docs/intro-react.md

```diff
@@ -0,0 +1,476 @@
+
+```SnackPlayer name=Your%20Cat
+import React from 'react';
+import { Text } from 'react-native';
+
+export default function Cat() {
+  return (
+    <Text>Hello, I am your cat!</Text>
+  );
+}
+```
+
+Here is how you do it: To define your `Cat` component, first use JavaScript’s [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) to import React and React Native’s [`Text`](/react-native/docs/next/text) Core Component:
+
+```jsx
+import React from 'react';
+import {Text} from 'react-native';
+```
+
+Your component starts as a function:
+
+```jsx
+function Cat() {}
+```
+
+Whatever a function component returns is rendered as a React element. `Cat` will render a `<Text>` element:
+
+```jsx
+function Cat() {
+  return <Text>Hello, I am your cat!</Text>;
+}
+```
+
+You can export your function component with JavaScript’s [`export`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) for use throughout your app like so:
+
+```jsx
+export default function Cat() {
+  // element(s) go here!
+}
+```
+
+<block class="classical syntax" />
+
+Class components have a little more complexity compared to function components in some ways.
+
+```SnackPlayer name=Your%20Cat
+import React, { Component } from 'react';
+import { Text } from 'react-native';
+
+export default class Cat extends Component {
+  render() {
+    return (
+      <Text>Hello, I am your cat!</Text>
+    );
+  }
+}
+```
+
+You additionally import `Component` from React:
+
+```jsx
+import React, {Component} from 'react';
+```
+
+Your component starts as a class extending `Component` instead of as a function:
+
+```jsx
+class Cat extends Component {}
+```
+
+Class components have a `render()` function. Whatever is returned inside it is rendered as a React element:
+
+```jsx
+class Cat extends Component {
+  render() {
+    return <Text>Hello, I am your cat!</Text>;
+  }
+}
+```
+
+And as with function components, you can export your class component:
+
+```jsx
+export default function Cat() {
+  // element(s) go here!
+}
+```
+
+<block class="endBlock syntax" />
+
+> This is one of many ways to export your component. This kind of export works well with the Snack Player. However, depending on your app’s file structure, you might need to use a different convention. This [handy cheatsheet on JavaScript imports and exports](https://medium.com/dailyjs/javascript-module-cheatsheet-7bd474f1d829) can help.
+
+Now take a closer look at that `return` statement. `<Text>Hello, I am your cat!</Text>` is using a kind of JavaScript syntax that makes writing elements convenient: JSX.
+
+## JSX
+
+React and React Native use JSX, a syntax that lets you write elements inside JavaScript like so: `<Text>Hello, I am your cat!</Text>`. The React docs have [a comprehensive guide to JSX](https://reactjs.org/docs/introducing-jsx.html) you can reference to learn even more. Here’s how you would write the previous example with JSX:
```

Comment:
So here's an interesting thing. The doc you link to is an intro to JSX. But we also *do* have a [more in-depth look at JSX](https://reactjs.org/docs/jsx-in-depth.html). Whether it's useful here or not is up to you. Maybe a link at the end.